core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala

```diff
@@ -176,8 +176,11 @@ class DockerContainer(protected val id: ContainerId,
   protected val waitForOomState: FiniteDuration = 2.seconds
   protected val filePollInterval: FiniteDuration = 5.milliseconds
 
-  def suspend()(implicit transid: TransactionId): Future[Unit] =
-    if (useRunc) { runc.pause(id) } else { docker.pause(id) }
+  override def suspend()(implicit transid: TransactionId): Future[Unit] = {
+    super.suspend().andThen {
+      case _ => if (useRunc) { runc.pause(id) } else { docker.pause(id) }
+    }
```

Comment:
We shouldn't use `andThen` here, because the Future should only complete after the pause operation has finished, `andThen` doesn't do that.

```scala
super.suspend().flatMap(_ => if(useRunc) runc.pause(id) else docker.pause(id))
```

Should be the way to go. We shouldn't ignore a failure of `super.suspend`, should we?